WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m

```diff
@@ -1328,14 +1328,19 @@ - (void)showMediaPicker
     options.badgedUTTypes = [NSSet setWithObject: (__bridge NSString *)kUTTypeGIF];
     options.preferredStatusBarStyle = UIStatusBarStyleLightContent;
     WPNavigationMediaPickerViewController *picker = [[WPNavigationMediaPickerViewController alloc] initWithOptions:options];
-    self.mediaDataSource = [[WPAndDeviceMediaLibraryDataSource alloc] initWithPost:self.apost
-                                                             initialDataSourceType:MediaPickerDataSourceTypeMediaLibrary];
-    picker.dataSource = self.mediaDataSource;
+
+    WPAndDeviceMediaLibraryDataSource *mediaDataSource = [[WPAndDeviceMediaLibraryDataSource alloc] initWithPost:self.apost
+                                                                                           initialDataSourceType:MediaPickerDataSourceTypeMediaLibrary];
+    mediaDataSource.mediaLibraryIgnoresNetworkLoadingErrors = YES;
```

Comment:
Not sure I understand what we're doing here. If we are always setting the mediaLibraryIgnoresNetworkLoadingErrors to true why even use it?

Reply:
`WPAndDeviceMediaLibraryDataSource` is also used in different places and I didn't want to silently change the behavior for other screens in the app

Reply:
Oh! I see it's used in site creation. What happens there when picking media in offline?

Reply:
the lower-level `MediaLibraryPickerDataSource` is also used in bunch of other places, like `GutenbergMediaPickerHelper` and `AztecPostViewController`, `PostEditor`, etc.

I don't want to change the behavior for those.

the `WPAndDeviceMediaLibraryDataSource` that wraps both local images and `MediaPicker` is also used in picking the site icon — it's actually a great point you raised and I changed it to work in both places correctly.